src/component/environments/EnvironmentActionCell/EnvironmentActionCell.tsx

```diff
@@ -0,0 +1,191 @@
+    environment: IEnvironment;
+}
+
+export const EnvironmentActionCell = ({
+    environment,
+}: IEnvironmentTableActionsProps) => {
+    const navigate = useNavigate();
+    const { hasAccess } = useContext(AccessContext);
+    const updatePermission = hasAccess(UPDATE_ENVIRONMENT);
+    const search = useSearchHighlightContext();
+
+    const { setToastApiError, setToastData } = useToast();
+    const { refetchEnvironments } = useEnvironments();
+    const { refetch: refetchPermissions } = useProjectRolePermissions();
+    const { deleteEnvironment, toggleEnvironmentOn, toggleEnvironmentOff } =
+        useEnvironmentApi();
+
+    const [deleteModal, setDeleteModal] = useState(false);
+    const [toggleModal, setToggleModal] = useState(false);
+    const [confirmName, setConfirmName] = useState('');
+
+    const handleDeleteEnvironment = async () => {
+        try {
+            await deleteEnvironment(environment.name);
+            refetchPermissions();
+            setToastData({
+                type: 'success',
+                title: 'Project environment deleted',
+                text: 'You have successfully deleted the project environment.',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        } finally {
+            setDeleteModal(false);
+            setConfirmName('');
+            await refetchEnvironments();
+        }
+    };
+
+    const handleConfirmToggleEnvironment = () => {
+        return environment.enabled
+            ? handleToggleEnvironmentOff()
+            : handleToggleEnvironmentOn();
+    };
+
+    const handleToggleEnvironmentOn = async () => {
+        try {
+            await toggleEnvironmentOn(environment.name);
+            setToggleModal(false);
+            setToastData({
+                type: 'success',
+                title: 'Project environment enabled',
+                text: 'Your environment is enabled',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        } finally {
+            await refetchEnvironments();
+        }
+    };
+
+    const handleToggleEnvironmentOff = async () => {
+        try {
+            await toggleEnvironmentOff(environment.name);
+            setToggleModal(false);
+            setToastData({
+                type: 'success',
+                title: 'Project environment disabled',
+                text: 'Your environment is disabled.',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        } finally {
+            await refetchEnvironments();
+        }
+    };
+
+    const toggleIconTooltip = environment.enabled
+        ? 'Disable environment'
+        : 'Enable environment';
+
+    const editId = useId();
+    const deleteId = useId();
+
+    return (
+        <ActionCell>
+            <ConditionallyRender
+                condition={updatePermission && !search}
```

Comment:
I can't actually tell what the case the boolean logic here describes at a glance. Might be more obvious to folks who have spent more time here but if not, a function that does this with a nice name might be better